File: A Problems/344A - Magnets/Magnets.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var scanner *bufio.Scanner
	var lines []string
	var previous string
	var groups int = 0

	scanner = bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	for i, magnet := range lines[1:] {
		if i == 0 {
			previous = magnet
			groups++
		} else {
			if magnet != previous {
				groups++
			}

			previous = magnet
		}
	}

	fmt.Println(groups)
}

File: A Problems/344A - Magnets/Magnets.pl
#!/usr/bin/perl

$n = <> + 0;
$groups = 1;
$previous = <>;

for ($i = 0; $i < ($n-1); $i++) {
    $magnet = <>;
    if ($magnet ne $previous) {
        $groups++;
    }

    $previous = $magnet;
}

print "$groups\n";